microsoft/testsuites/ltp/ltp.py

```diff
@@ -0,0 +1,355 @@
+    ) -> List[LtpResult]:
+        # tests cannot be empty
+        assert_that(ltp_tests, "ltp_tests cannot be empty").is_not_empty()
+        ls = self.node.tools[Ls]
+        rm = self.node.tools[Rm]
+
+        # remove skipfile if it exists
+        if ls.path_exists(self.LTP_SKIP_FILE):
+            self._log.debug(f"Removing skipfile: {self.LTP_SKIP_FILE}")
+            rm.remove_file(self.LTP_SKIP_FILE, sudo=True)
+
+        # remove results file if it exists
+        if ls.path_exists(self.LTP_RESULT_PATH, sudo=True):
+            self._log.debug(f"Removing {self.LTP_RESULT_PATH}")
+            rm.remove_file(self.LTP_RESULT_PATH, sudo=True)
+
+        # remove output file if it exists
+        if ls.path_exists(self.LTP_OUTPUT_PATH, sudo=True):
+            self._log.debug(f"Removing {self.LTP_OUTPUT_PATH}")
+            rm.remove_file(self.LTP_OUTPUT_PATH, sudo=True)
+
+        # add parameters for the test logging
+        parameters = f"-p -q -l {self.LTP_RESULT_PATH} -o {self.LTP_OUTPUT_PATH} "
+
+        # add the list of tests to run
+        parameters += f"-f {','.join(ltp_tests)} "
+
+        # add logging and output file parameter
+        if drive_name:
+            parameters += f"-z {drive_name} "
+
+        # add the list of skip tests to run
+        if len(skip_tests) > 0:
+            # write skip test to skipfile with newline separator
+            skip_file_value = "\n".join(skip_tests)
+            self.node.tools[Echo].write_to_file(
+                skip_file_value, PurePosixPath(self.LTP_SKIP_FILE), sudo=True
+            )
+            parameters += f"-S {self.LTP_SKIP_FILE} "
+
+        # Minimum 4M swap space is needed by some mmp test
+        if self.node.tools[Free].get_swap_size() < 4:
+            self.node.tools[Swap].create_swap()
+
+        # run ltp tests
+        self.run(parameters, sudo=True, force_run=True, timeout=12000)
+
+        # write output to working_path
+        cat = self.node.tools[Cat]
+        with open(f"{log_path}/ltp-output.log", "w") as f:
+            f.write(
+                cat.read(
+                    self.LTP_OUTPUT_PATH, sudo=True, force_run=True, no_debug_log=True
+                )
+            )
+
+        # write results to working_path
+        with open(f"{log_path}/ltp-results.log", "w") as f:
+            f.write(
+                cat.read(
+                    self.LTP_RESULT_PATH, sudo=True, force_run=True, no_debug_log=True
+                )
+            )
+
+        # parse results
+        results = self._parse_results(self.LTP_RESULT_PATH)
+
+        # assert that all tests passed
+        failed_tests = []
+        for result in results:
+            if result.status == TestStatus.FAILED:
+                failed_tests.append(result.name)
+
+            # create test result message
+            info: Dict[str, Any] = {}
+            info["information"] = {}
+            info["information"]["architecture"] = result.architecture
```

Comment:
The architecture should be common for all community test cases. Use the value from kernel information of os, so it can be shared cross test cases.